File: Python/Rebill/Create_Recurring_Payment_ACH.py
##
# BluePay Python Sample code.
#
# This code sample creates a recurring payment charging $15.00 per month for one year.
##

import os.path, sys
sys.path.append(os.path.join(os.path.dirname(os.path.realpath(__file__)), os.pardir))
from BluePay import BluePay

account_id = "100228390579"
secret_key = "AKGIF9X9WT9CLQCWDFONC8N3HXRL9Y5K"
mode = "TEST"

payment = BluePay(
    account_id = account_id, # Merchant's Account ID
    secret_key = secret_key, # Merchant's Secret Key
    mode = mode # Transaction Mode: TEST (can also be LIVE)
)

# Set Customer Information
payment.set_customer_information(
    name1 = "Bob",
    name2 = "Tester",
    addr1 = "123 Test St.",
    addr2 = "Apt #500",
    city = "Testville",
    state = "IL",
    zipcode = "54321",
    country = "USA"
)

# Set ACH Payment Information
payment.set_ach_information(
    routing_number = "123123123", # Routing Number: 123123123
    account_number = "05125121", # Account Number: 0523421
    account_type = "C", # Account Type: Checking
    doc_type = "WEB" # ACH Document Type: WEB
)

payment.auth(amount = '0.00')

# Set Recurring Payment Information
payment.set_rebilling_information(
  reb_first_date = "2015-01-01", # Rebill Start Date: Jan. 1, 2015
  reb_expr = "1 MONTH", # Rebill Frequency: 1 MONTH
  reb_cycles = "12", # Rebill # of Cycles: 12
  reb_amount = "15.00" # Rebill Amount: $15.00
)

# Makes the API Request
payment.process()

# Read response from BluePay
if payment.is_successful_response():
    print 'Transaction ID: ' + payment.trans_id_response
    print 'Rebill ID: ' + payment.reb_id_response
    print 'Transaction Status: ' + payment.status_response
    print 'Transaction Message: ' + payment.message_response
    print 'Masked Payment Account: ' + payment.trans_id_response
    print 'Customer Bank: ' + payment.bank_name_response
else:
    print payment.message_response

File: PHP/Rebill/Create_Recurring_Payment_ACH.php
<?php
/**
* BluePay PHP Sample Code
* This code sample creates a recurring payment charging $15.00 per month for one year.
*/

include('../BluePay.php');

$accountID = "100228390579";
$secretKey = 'AKGIF9X9WT9CLQCWDFONC8N3HXRL9Y5K';
$mode = 'TEST';

$rebill = new BluePay(
    $accountID,
    $secretKey,
    $mode
);

$rebill->setCustomerInformation(array(
    'firstName' => 'Bob', 
    'lastName' => 'Tester', 
    'addr1' => '1234 Test St.', 
    'addr2' => 'Apt #500', 
    'city' => 'Testville', 
    'state' => 'IL', 
    'zip' =>'54321', 
    'country' => 'USA', 
    'phone' => '1231231234', 
    'email' => 'test@bluepay.com' 
));

$rebill->setACHInformation(array(
    'routingNumber' =>'123123123', // Routing Number: 123123123
    'accountNumber' => '1234567890', // Account Number: 1234567890
    'accountType' => 'C', // Account Type: Checking
    'documentType' => 'WEB' // ACH Document Type: WEB
));

$rebill->setRebillingInformation(array(
   'rebillFirstDate' => '2015-01-01', // Rebill Start Date: Jan. 1, 2015
   'rebillExpression' => '1 MONTH', // Rebill Frequency: 1 MONTH
   'rebillCycles' => '12', // Rebill # of Cycles: 12
   'rebillAmount' => '15.00' // Rebill Amount: $15.00
));

$rebill->auth('0.00');

# Makes the API Request with BluePay
$rebill->process();

if ($rebill->isSuccessfulResponse()) {
    # Read response from BluePay
    echo 
    'Transaction ID: '. $rebill->getTransID() . "\n" .
    'Rebill ID: ' . $rebill->getRebillID() . "\n" .
    'Status: '. $rebill->getStatus() . "\n" .
    'Message: '. $rebill->getMessage() . "\n" .
    'Masked Account: ' . $rebill->getMaskedAccount() . "\n" .
    'Customer Bank: ' . $rebill->getBank() . "\n";
} else {
    echo $rebill->getMessage();
}

?>